plugins/kettle-xml-plugin/src/org/pentaho/di/trans/steps/getxmldata/GetXMLData.java

```diff
@@ -93,6 +96,14 @@ protected boolean setDocument( String StringXML, FileObject file, boolean IsInXM
       // Validate XML against specified schema?
       if ( meta.isValidating() ) {
         reader.setValidation( true );
+        try {
+          reader.setFeature( XMLConstants.FEATURE_SECURE_PROCESSING, true );
```

Comment:
This is repetitive code here - can it (and the ones like it below) be moved to a Utils class or static methods of XMLParserFactoryProducer so that it isn't repeated over and over. Eventually, there will be another security-focused setting that will need to be set, and we'll have to touch all these files. I'd rather have them all call a utility method and have the utility method be the only one responsible for knowing all the features.